customerio/customerio-ios
# feat: display an inline message in a SwiftUI app

PR for https://linear.app/customerio/issue/MBL-402/display-an-inline-message-in-a-swiftui-app

### Problem:
As a SwiftUI developer, I want to display an inline in-app message within my SwiftUI app so that I can provide users with seamless in-app messaging experiences that are consistent with those in UIKit.

### Solution:
As UIKit views are not directly supported in SwiftUI apps hence we are using `UIViewRepresentable` to make use of `InAppMessageView` in the SwiftUI sample app. 
 
With minimal updates, add an inline message to the SwiftUI view and expect the same behaviour as a UIKit sample app.

### Demo:
- Close message and display next message in queue
- Persistent messages
- Action buttons - Close, custom
https://github.com/user-attachments/assets/2869ea46-44d5-4e7e-9b88-88fc22fcc376

- show another message
https://github.com/user-attachments/assets/153d0d79-7e84-4670-bc74-7b6be711eaf0

- deep links 
https://github.com/user-attachments/assets/0af567b6-5ab4-422e-85b4-9b377927091e

### Testing:
For manual testing, send an inline in-app message to the device
- Once it displays, check activity log events
- Tap on action button on inline message
- Check activity log for clicked callback
- Tap on close button to see activity indicator and next message in queue
- Perform same tests like on UIKit sample app
